File: src/medium/array-stack.ts
export const arrayStack = (target: number[], n: number): string[] => {
    const result: string[] = [];
    let counter = 1;
    let index = 0;
    while (index < target.length) {
        result.push("Push");
        if (target[index] === counter) {
            index++;
        } else {
            result.push("Pop");
        }
        counter++;
    }
    return result;
};

File: lib/medium/array_stack.dart
List<String> arrayStack(List<int> target, int n) {
    List<String> result = [];
    int index = 0;
    int counter = 1;
    while (index < target.length) {
      result.add("Push");
      if (target[index] == counter) {
        index++;
      } else {
        result.add("Pop");
      }
      counter++;
    }
    return result;
}
